pgjdbc/src/test/java/org/postgresql/test/jdbc42/UpdateObject310Test.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.jdbc42;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PSQLException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.TimeZone;
+
+@RunWith(Parameterized.class)
+public class UpdateObject310Test {
+  private static final TimeZone saveTZ = TimeZone.getDefault();
+
+  private Connection con;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+        {LocalDateTime.now().truncatedTo(ChronoUnit.MICROS), "timestamp_without_time_zone_column",
+            "'2003-01-01'::TIMESTAMP",
+            "ERROR: column \"timestamp_without_time_zone_column\" is of type timestamp without time zone but expression is of type integer\n"
+                + "  Hint: You will need to rewrite or cast the expression.\n"
+                + "  Position:"},
+        {LocalDate.now(), "date_column", "'2003-01-01'::DATE",
+            "ERROR: column \"date_column\" is of type date but expression is of type integer\n"
+                + "  Hint: You will need to rewrite or cast the expression.\n"
+                + "  Position:"},
+        {LocalTime.now().truncatedTo(ChronoUnit.MICROS), "time_without_time_zone_column",
+            "'07:23'::TIME WITHOUT TIME ZONE",
+            "ERROR: column \"time_without_time_zone_column\" is of type time without time zone but expression is of type integer\n"
+                + "  Hint: You will need to rewrite or cast the expression.\n"
+                + "  Position:"}
+    });
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    con = TestUtil.openDB();
+    TestUtil.createTableWithPrimaryKey(con, "table1", "timestamp_without_time_zone_column timestamp without time zone,"
+        + "date_column date,"
+        + "time_without_time_zone_column time without time zone"
+    );
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    TimeZone.setDefault(saveTZ);
```

Comment:
If the dedicated `java.time` methods from `TimestampUtils` are used this should no longer be necessary.